### PR TITLE
test.cross.sanity: init

### LINKED ISSUE #243320: we need a `pkgsCross` smoke test
Until recently, nixpkgs' cross-compilation infrastructure was considered a sort of exotic/best-effort thing, without much CI or testing until things got to Hydra.  That's no longer the case, and we are starting to depend on `pkgsCross` for increasingly important stuff.

We need a package-sized (i.e. not needing `hydra-eval-jobs`) build that people can run when refactoring our cross-compilation infrastructure.

When I wrote https://github.com/NixOS/nixpkgs/pull/238154 I tested it thouroughly on all the `targetPlatform`s I encounter on a regular basis, but since I don't use LLVM much or Windows at all I neglected to try `pkgsLLVM` and `pkgsCross.<various-windows-flavors>`.  [That's not cool](https://github.com/NixOS/nixpkgs/pull/238154#issuecomment-1633752094).

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

This commit adds `pkgs.test.cross.sanity`, which is meant to be a carefully curated list of builds/packages that tend to break when refactoring our cross-compilation infrastructure.

It should strike a balance between being small enough to fit in a single eval (i.e. not so large that hydra-eval-jobs is needed) so we can ask @ofborg to check it, yet should have good examples of things that often break.  So, no buckshot `mapTestOnCross` calls here.

Closes #243320

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build test.cross.sanity

**Comment by Artturin** [comment]
@ofborg build tests.cross.sanity

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build tests.cross.sanity

**Comment by amjoseph-nixpkgs** [comment]
tests.cross.sanity on x86_64-linux — Success [Details](https://github.com/NixOS/nixpkgs/pull/243321/checks?check_run_id=15036947174)

File: pkgs/test/cross/default.nix [hunk 1]
```diff
@@ -107,7 +107,43 @@ let
...
+  sanity = [
+    #pkgs.mbuffer              # https://github.com/NixOS/nixpkgs/issues/213453
+    #pkgs.pkgsCross.gnu64.bash # https://github.com/NixOS/nixpkgs/issues/243164
+    pkgs.gcc_multi.cc
+    pkgs.pkgsMusl.stdenv
+    pkgs.pkgsLLVM.stdenv
+    pkgs.pkgsStatic.bash
+    pkgs.pkgsCross.arm-embedded.stdenv
+    pkgs.pkgsCross.aarch64-multiplatform.pkgsBuildTarget.gcc
+    #pkgs.pkgsCross.powernv.pkgsBuildTarget.gcc
```
Comment (amjoseph-nixpkgs) [inline pkgs/test/cross/default.nix:138]: I think I know how to fix this, but I want to make sure I don't break native builds in doing so.  My PowerPC hardware is offline until Monday, so for now I've just disabled this.

**Comment by amjoseph-nixpkgs** [comment]
Rebased.

**Pull request merged**